src/System.Buffers.Experimental/System/Buffers/Text/BufferWriter_strings.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            while (!TryWriteLine(text, format)) Resize();
+        }
+        #endregion
+
+        #region UTF8String
+        public bool TryWrite(Utf8Span text)
+        {
+            int written = text.Bytes.Length;
+            if (!text.Bytes.TryCopyTo(Free)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(Utf8Span text)
+        {
+            while (!TryWrite(text)) Resize();
+        }
+
+        public bool TryWrite(Utf8Span text, TransformationFormat format)
+        {
+            int written = text.Bytes.Length;
+            if (!text.Bytes.TryCopyTo(Free)) return false;
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(Utf8Span text, TransformationFormat transformation)
+        {
+            while (!TryWrite(text, transformation)) Resize();
+        }
+
+        public bool TryWriteLine(Utf8Span text)
+        {
+            if (!TryWriteLine(text, out var written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(Utf8Span text)
+        {
+            while (!TryWriteLine(text)) Resize();
+        }
+
+        public bool TryWriteLine(Utf8Span text, TransformationFormat format)
+        {
+            if (!TryWriteLine(text, out var written)) return false;
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(Utf8Span text, TransformationFormat transformation)
+        {
+            while (!TryWriteLine(text, transformation)) Resize();
+        }
+
+        public bool TryWrite(Utf8String text)
+            => TryWrite((Utf8Span)text);
+
+        public void Write(Utf8String text)
+            => Write((Utf8Span)text);
+
+        public bool TryWrite(Utf8String text, TransformationFormat transformation)
+            => TryWrite((Utf8Span)text, transformation);
+
+        public void Write(Utf8String text, TransformationFormat transformation)
+            => Write((Utf8Span)text, transformation);
+        #endregion
+
+        public void Write(char character)
+        {
+            if (character <= 127)
+            {
+                var free = Free;
+                if (free.Length < 1)
+                {
+                    Resize();
+                    free = Free;
+                }
+                free[0] = (byte)character;
+                _written++;
+            }
+            else
+            {
+                Span<byte> utf8Span = stackalloc byte[4];
+                Span<char> utf16Span = stackalloc char[1];
+                utf16Span[0] = character;
+                if(Encodings.Utf8.FromUtf16(utf16Span.AsBytes(), utf8Span, out int consumed, out int written) == OperationStatus.Done)
```

Comment:
The call to AsBytes is likely too expensive here.

Can we use bitwise operations and get the two bytes that represent the character ourselves and change utf16Span to the following?
```C#
Span<byte> utf16ByteSpan = stackalloc byte[2];
```

Reply:
Yeah, I forgot to add a comment that this is a naive implementation. I think we need char encoding APIs on Utf8/Utf16 classes. But I dont think char fits in 2 bytes. I think it fits in 3, but I wanted to err on the safe side (especially for a implementation that will be optimized later)